src/adapter/src/coord/command_handler.rs

```diff
@@ -280,6 +280,7 @@ impl Coordinator {
                 secret_key: session.secret_key(),
                 notice_tx: session.retain_notice_transmitter(),
                 drop_sinks: Vec::new(),
+                current_role: session.current_role_id().clone(),
```

Comment:
We should probably use the session role since it's closer to the authenticated role and leave a TODO saying switch this to the authenticated role when it's implemented.

Reply:
How does authenticated role differ from session role? Aren't those both set once at connection start?

Reply:
The session role can be modified via `SET SESSION AUTHORIZATION` (https://www.postgresql.org/docs/15/sql-set-session-authorization.html). The authenticated role will never change.

Reply:
Ohh I didn't know this was a thing. So there's both `SET ROLE` and `SET SESSION AUTHORIZATION`? Neat?

Reply:
The authenticated role is used to determine privileges for:
- `SET SESSION AUTHORIZATION`.
- system admin functions, like `pg_cancel_backend`.
- Maybe others that I don't know about.

The session role is used to determine privileges for:
- `SET ROLE`/`RESET ROLE`.
- Maybe others that I don't know about.